CIP-0027.md

```diff
@@ -0,0 +1,64 @@
+
+## Abstract
+
+This proposed standard will allow for uniform royalties&#39; distributions across the secondary market space. It is easy to implement using metadata only, and does not require a smart contract.  However, it is scalable to allow for the usage of a downstream smart contract, as needed by the asset creator. It has been developed with input from Artano, BuffyBot, CNFT.io, Digital Syndicate, Fencemaker, MADinArt, NFT-Maker.io, Hydrun, Tokhun, and many more.
+
+## Motivation
+
+There is a significant interest within the Cardano Community for an implementation of royalties distribution when a Cardano Asset is resold on the secondary market. It has become a common theme to see and hear statements that the only thing stopping artists from adopting Cardano, is that they are waiting for an implementation of royalties.  At the present time, smart contracts do not create a simple mechanism to implement royalties.  By developing a community standard, we can resolve the immediate need for royalties, and create a path forward for a potential future iteration of smart contracts.
+
+## Specification
+
+A new tag of 777 is proposed for this implementation.  The community guidelines have been agreed as follows:
+1) The royalties tags are to be written to an unnamed token, using the policy to be used for the intended Cardano Assets.
+2) Only the first minted set of instructions will be honored.  Any future updates or rewrites will be ignored.  This prevents a Cardano Asset maker from changing the royalties at a future date.
+3) Within this created asset will be the metadata for royalties distributions.  It will use a tag of 777, and then have two tags to identify the percentage of future sales requested as a royalty, and the payment address to forward those royalties to.  Those tags will be "pct" and "addr" respectively.
+4) The "pct" key tag can be any floating point value from 0.0 to 1.0, to represent between 0 and 100 percent.  For example, a 12.5 percent royalty would be represented with "pct": "0.125".
+5) The "addr" key tag can be a string value, or an array.  It is to include a single payment address.  By allowing for an array, the payment address can exceed the per line 64 character limitation.  This payment address could be part of a smart contract, which should allow for greater flexibility of royalties distributions, controlled by the asset creator.
+6) The royalty token must be minted prior to creating any assets off the policy.  All markets will be instructed to look at the timeline of use on a policy.  Any assets minted prior to the royalty token would not be honored.
+
+## Example JSON with string
+
+{
+	"777": {
+		"pct": "0.2",
+		"addr": "addr1v9nevxg9wunfck0gt7hpxuy0elnqygglme3u6l3nn5q5gnq5dc9un"
+	}
+}
+
+## Example JSON with array
+
+{
+	"777": {
+		"pct": "0.2",
+		"addr": [
+			"addr1q8g3dv6ptkgsafh7k5muggrvfde2szzmc2mqkcxpxn7c63l9znc9e3xa82h",
```

Comment:
Is there a benefit to using bech32 here? You could save space by just saving a byte array for the address. The benefit of bech32 is the checksum and the fact it's more human readable, but this CIP is mostly for machines to read, not humans.

Reply:
On the other hand, humans will be crafting this metadata when minting their assets. Seems like a useful place for checksum-ing.

Reply:
> Is there a benefit to using bech32 here? You could save space by just saving a byte array for the address. The benefit of bech32 is the checksum and the fact it's more human readable, but this CIP is mostly for machines to read, not humans.

How to make a byte-array in a metadata.json for a transaction? This is not a supported format currently.

Reply:
> > Is there a benefit to using bech32 here? You could save space by just saving a byte array for the address. The benefit of bech32 is the checksum and the fact it's more human readable, but this CIP is mostly for machines to read, not humans.
> 
> How to make a byte-array in a metadata.json for a transaction? This is not a supported format currently.

Hex encoded bytestrings are supported when hex-encoded (and prefaced with `0x`), please [see documentation here](https://developers.cardano.org/docs/transaction-metadata/).

Reply:
Ok, i learned something today:

1.) the documentation example is wrong 😄 
2.) looks like this is really working. strange, now i have to rethink some other strings that me be used.

Reply:
i dont think `bech32` encoding is a good idea here as @huths0lo remarked.`bech32` encoding is quite a complex process for non-technical folk. in addition, because of the strict v. loose encoding options along with various versions of addresses (with stake info and without) there's too much room for error and likely to ultimately hurt adoption.